Sources/SourceKitLSP/Swift/OpenInterface.swift

```diff
@@ -0,0 +1,85 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2022 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import Foundation
+import SourceKitD
+import LanguageServerProtocol
+import LSPLogging
+
+struct InterfaceInfo {
+  var contents: String
+}
+
+private var textualInterfaces: [String:DocumentURI] = [:]
+
+extension SwiftLanguageServer {
+  public func openInterface(_ request: LanguageServerProtocol.Request<LanguageServerProtocol.OpenInterfaceRequest>) {
+    let uri = request.params.textDocument.uri
+    let moduleName = request.params.name
+    self.queue.async {
+      
+      if let uri = textualInterfaces[moduleName] {
```

Comment:
I don’t think we can cache textual interfaces like this. The contents of the module might have changed since it was last requested and thus the cache would be out of date. 

I think caching would be safe for system modules that are part of the SDK and are thus very, very unlikely to change (unless you re-install Xcode during a session). IIRC there should be a bit in either the symbol or the module response that says whether the module is a system module.

Also the compiler arguments would need to be part of the key because the generated interface request can return different results for different compiler arguments (think about the generated interface from an Objective-C module that uses `#if FOO` and in one invocation you pass `-DFOO` in the compiler arguments and in the other you don’t.

Reply:
I removed caching for now. We could reintroduce it later while addressing the issues you raised.

Reply:
OK, I think even without caching this is a very good PR.